src/platform/Linux/DiagnosticDataProviderImpl.cpp

```diff
@@ -218,28 +218,33 @@ DiagnosticDataProviderImpl & DiagnosticDataProviderImpl::GetDefaultInstance()
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapFree(uint64_t & currentHeapFree)
 {
+#ifdef __GLIBC__
     struct mallinfo mallocInfo = mallinfo();
 
     // Get the current amount of heap memory, in bytes, that are not being utilized
     // by the current running program.
     currentHeapFree = mallocInfo.fordblks;
```

Comment:
should there not be an else value that does something?

Like return CHIP_ERROR_NOT_IMPLEMENTED or at least setting the current free to some know value.

Reply:
sure - will change to something like...

```cpp
#ifndef __GLIBC__
    return CHIP_ERROR_NOT_IMPLEMENTED;
#else
...
#endif
```